File: sol/solution/0001-0100/0036/Solution.swift
class Solution {
    func isValidSudoku(_ board: [[Character]]) -> Bool {
        var rows = [Int](repeating: 0, count: 9)
        var cols = [Int](repeating: 0, count: 9)
        var boxes = [Int](repeating: 0, count: 9)
        for i in 0..<9 {
            for j in 0..<9 {
                let c = board[i][j]
                if c == "." { continue }
                let v = Int(c.asciiValue! - Character("1").asciiValue!)
                let mask = 1 << v
                let b = (i / 3) * 3 + (j / 3)
                if (rows[i] & mask) != 0 { return false }
                if (cols[j] & mask) != 0 { return false }
                if (boxes[b] & mask) != 0 { return false }
                rows[i] |= mask
                cols[j] |= mask
                boxes[b] |= mask
            }
        }
        return true
    }
}

File: sol/solution/0001-0100/0036/Solution.rb
# @param {Character[][]} board
# @return {Boolean}
def is_valid_sudoku(board)
  rows = Array.new(9, 0)
  cols = Array.new(9, 0)
  boxes = Array.new(9, 0)
  (0...9).each do |i|
    (0...9).each do |j|
      c = board[i][j]
      next if c == '.'
      v = c.ord - '1'.ord
      mask = 1 << v
      b = (i / 3) * 3 + (j / 3)
      return false if (rows[i] & mask) != 0
      return false if (cols[j] & mask) != 0
      return false if (boxes[b] & mask) != 0
      rows[i] |= mask
      cols[j] |= mask
      boxes[b] |= mask
    end
  end
  true
end